src/main/java/io/github/darealturtywurty/turtychemistry/core/multiblocks/MultiblockListener.java

```diff
@@ -17,44 +17,54 @@
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 import net.minecraftforge.fml.common.Mod;
 import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.Level;
 import org.jetbrains.annotations.Nullable;
 
 @Mod.EventBusSubscriber(modid = TurtyChemistry.MODID, bus = Mod.EventBusSubscriber.Bus.FORGE)
 public final class MultiblockListener {
     @SubscribeEvent
     public static void blockPlace(BlockEvent.EntityPlaceEvent event) {
+        //cache the event caller's Level and Block Position
+
         if(!(event.getEntity() instanceof Player) || event.getLevel().isClientSide())
             return;
-
+        final LevelAccessor currentEventLevel = event.getLevel();
+        final BlockPos currentEventPosition = event.getPos();
+        final BlockState currentEventPlacedBlock = event.getPlacedBlock();
         for(Multiblock multiblock : MultiblockRegistry.REGISTRY.get()) {
-            if(!multiblock.isValid(event.getPlacedBlock())) {
-                continue;
-            } else {
+            if(multiblock.isValid(currentEventPlacedBlock)) {
```

Comment:
Why is it like this? If its not valid it should continue.

Reply:
because intrinsically if the multiblock isn't valid it'll still loop until it is valid

Reply:
Well yes but a different multiblock _could_ be valid.

Reply:
The way it was done before means that you are just avoiding the unnecessary indentation.

Reply:
either way it's still looping through every possible multiblock present in the multiblock registry, all this condition did was say if the multiblock isn't valid continue to loop, which instead of checking if it's not valid then doing the checks, the loop will still continue until the multiblock is valid, in the multiblock check as well we're returning the method

Reply:
What? You realise it's exactly the same code right? It doesn't function any differently. The only difference is that you have added an extra indentation.